package MsgsClientToAuthentication;

import "AuthenticationToClient.proto";
import "Version.proto";

message MsgAuthenticationRequest {
   required string login    = 1;
   required string password = 2; 
   required Version.ProtocolVersion version = 3;
}

message MsgProfileCreationRequest {
    required MsgsAuthenticationToClient.MsgProfile profile = 1;
}

message MsgProfileUpdateRequest {
    required string login = 1;
    required MsgsAuthenticationToClient.MsgProfile profile = 2;
}

message MsgProfileDestructionRequest {
    required string login = 1;
}

message MsgClientToAuthentication {
    message Content {
       optional MsgAuthenticationRequest     authentication_request      = 1;
       optional MsgProfileCreationRequest    profile_creation_request    = 2;
       optional MsgProfileUpdateRequest      profile_update_request      = 3;
       optional MsgProfileDestructionRequest profile_destruction_request = 4;
    }
    optional int32   context = 1 [default = 0];
    required Content message = 2;
}
